pkg/azuredisk/controllerserver.go

```diff
@@ -518,51 +518,70 @@ func (d *Driver) ListVolumes(ctx context.Context, req *csi.ListVolumesRequest) (
 		}
 	}
 
-	disks, derr := d.cloud.DisksClient.ListByResourceGroup(ctx, d.cloud.ResourceGroup)
-	if derr != nil {
-		return nil, status.Error(codes.Internal, fmt.Sprintf("ListVolumes on rg(%s) failed with error: %v", d.cloud.ResourceGroup, derr.Error()))
+	if d.cloud.KubeClient == nil || d.cloud.KubeClient.CoreV1() == nil || d.cloud.KubeClient.CoreV1().PersistentVolumes() == nil {
```

Comment:
if kubeClient is not set, should return `DisksClient.ListByResourceGroup` result instead of error